src/keboola_mcp_server/tools/components/model.py

```diff
@@ -1,354 +1,550 @@
-            'configurationDescription',
-            'configuration-description',
-        ),
-        serialization_alias='configurationDescription',
-        default=None,
+        description='Component ID',
     )
-    is_disabled: bool = Field(
-        description='Whether the component configuration is disabled',
-        validation_alias=AliasChoices('isDisabled', 'is_disabled', 'is-disabled'),
-        serialization_alias='isDisabled',
-        default=False,
+    component_name: str = Field(
+        description='Component name',
     )
-    is_deleted: bool = Field(
-        description='Whether the component configuration is deleted',
-        validation_alias=AliasChoices('isDeleted', 'is_deleted', 'is-deleted'),
-        serialization_alias='isDeleted',
-        default=False,
+    component_type: str = Field(
+        description='Component type',
     )
+    capabilities: ComponentCapabilities = Field(description='Component capabilities')
 
+    @classmethod
+    def from_api_response(cls, api_response: APIComponentResponse) -> 'ComponentSummary':
+        """
+        Create ComponentSummary from API response.
 
-class Component(ReducedComponent):
+        :param api_response: Parsed API response from Storage or AI Service API
+        :return: Lightweight component domain model for list operations
+        """
+        capabilities = ComponentCapabilities.from_flags(api_response.flags)
+
+        return cls.model_construct(
+            component_id=api_response.component_id,
+            component_name=api_response.component_name,
+            component_type=api_response.type,
+            capabilities=capabilities,
+        )
+
+
+class Component(BaseModel):
     """
-    A Component containing detailed information about the Keboola Component, including its capabilities,
-    documentation, and configuration schemas.
+    Complete component representation with full details.
+
+    Contains comprehensive component information including documentation,
+    configuration schemas, and metadata. Used by get tools where detailed
+    component information is needed.
     """
 
+    # Core component metadata (shared with ComponentSummary)
+    component_id: str = Field(
+        description='Component ID',
+    )
+    component_name: str = Field(
+        description='Component name',
+    )
+    component_type: str = Field(
+        description='Component type',
+    )
     component_categories: list[str] = Field(
         default_factory=list,
-        description='The categories the component belongs to.',
-        validation_alias=AliasChoices(
-            'componentCategories', 'component_categories', 'component-categories', 'categories'
-        ),
-        serialization_alias='categories',
+        description='Component categories',
     )
-    documentation_url: Optional[str] = Field(
+    capabilities: ComponentCapabilities = Field(description='Component capabilities')
+
+    # Detailed metadata (not in ComponenSummary)
```

Comment:
```suggestion
    # Detailed metadata (not in ComponentSummary)
```
a spell-checker catches this